browser/resources/settings/email_aliases_page/tests/email_aliases.test.tsx

```diff
@@ -0,0 +1,173 @@
+    const mockEmailAliasesService = new MockEmailAliasesServiceInterface()
+    await act(async () => {
+      render(<ManagePage
+               emailAliasesService={mockEmailAliasesService}
+               bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+    expect(document.querySelector('leo-progressring'))
+      .toBeInTheDocument()
+    expect(screen.getByText(localeRegex(
+      'emailAliasesConnectingToBraveAccount'))).toBeInTheDocument()
+  })
+
+  it('shows sign up form when no email is available', async () => {
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onLoggedOut()
+    })
+    await act(async () => {
+      render(<ManagePage
+               emailAliasesService={mockEmailAliasesService}
+               bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(localeRegex('emailAliasesSignInOrCreateAccount')))
+        .toBeInTheDocument()
+      expect(screen.getByPlaceholderText(localeRegex(
+        'emailAliasesEmailAddressPlaceholder'))).toBeInTheDocument()
+    })
+  })
+
+  it('shows main view when email is available', async () => {
+    const mockEmail = 'test@brave.com'
+    const mockAliases = [
+      {
+        email: 'alias1@brave.com',
+        note: 'Test Alias 1',
+        domains: undefined,
+      }
+    ]
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onLoggedIn(mockEmail)
+      observer.onAliasesUpdated(mockAliases)
+    })
+
+    await act(async () => {
+      render(<ManagePage emailAliasesService={mockEmailAliasesService}
+            bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(mockEmail)).toBeInTheDocument()
+      expect(screen.getByText(/alias1@brave\.com/)).toBeInTheDocument()
+      expect(screen.getByText(/Test Alias 1/)).toBeInTheDocument()
+    })
+  })
+
+  it('shows verification pending view', async () => {
+    const mockEmail = 'test@brave.com'
+    const mockEmailAliasesService =
+      new MockEmailAliasesServiceInterface((observer) => {
+      observer.onVerificationPending(mockEmail)
+    })
+
+    await act(async () => {
+      render(<ManagePage emailAliasesService={mockEmailAliasesService}
+            bindObserver={createBindObserver(mockEmailAliasesService)}
+      />)
+    })
+
+    await waitFor(() => {
+      expect(screen.getByText(localeRegex('emailAliasesLoginEmailOnTheWay')))
+        .toBeInTheDocument()
+      expect(screen.getByText(localeRegex('emailAliasesClickOnSecureLogin')))
+        .toBeInTheDocument()
+      expect(screen.getByText(localeRegex('emailAliasesDontSeeEmail')))
+        .toBeInTheDocument()
+    })
+  })
+
+  it('triggers logout when sign out button is clicked', async () => {
```

Comment:
per below shouldn't we also verify that any existing email aliases are shown for login and hidden when logged out?

Reply:
I have expanded unit tests. They now include existing aliases being displayed.